src/resources/marketplace.js

```diff
@@ -108,6 +113,46 @@ class Marketplace extends Adaptable {
     })
   }
 
+  async search(searchQuery, listingType, filters = []) {
+
+
+    const query = `
+    {
+      listings (
+        searchQuery: "${searchQuery}"
+        filters: [${filters.map(filter => {
+          return `
+          {
+            name: "${filter.name}"
+            value: "${String(filter.value)}"
+            valueType: ${filter.valueType}
+            operator: ${filter.operator}
+          }
+          `
+        }).join(',')}]
+      ) {
+        pageNumber
+        itemsPerPage
+        totalNumberOfPages
+        nodes {
+          id
+
+        }
+      }
+    }`
+
+    return this.fetch(`${this.apolloServer}:${this.apolloServerPort}`, {
```

Comment:
Do you need to call .json() on the response to parse the response ? Something like:
```
const resp = this.fetch(...)
return resp.json()
```

Reply:
Http request might fail with different sort of errors. Like server not reachable, server returned 500 error, JSON was malformed.... And for each of those errors we would need to catch them separately in `origin-js` and re-throw them, so that client (dapp in this case) can show the correct error message. 

My logic was just return the promise, and let client handle different sorts of errors anyway it wishes. 

Can you maybe think of a way to better handle this?